### ISSUE TITLE
Seq.splitAt implementation is inefficient

### ISSUE THREAD

**Issue by tlinkowski** [jOOQ/jOOL#308]
Iterating `Seq.of(1, 2, 3, 4, 5, 6).splitAt(2).v1()` requires consumption of entire `Seq`, and not only consumption of the first 2 elements (i.e. until given `position` is reached).

I propose the following implementation using `SeqBuffer` proposed in #305:
```java
    static <T> Tuple2<Seq<T>, Seq<T>> splitAt(Stream<? extends T> stream, long position) {
        SeqBuffer<T> buffer = SeqBuffer.of(stream);
        return tuple(buffer.seq().limit(position), buffer.seq().skip(position));
    }
```

**Comment by lukaseder** [comment]
OK this depends on `SeqBuffer`. I'll look at that first: https://github.com/jOOQ/jOOL/issues/305

**Comment by lukaseder** [comment]
Hmm, I think you haven't sent a PR for these changes yet. Would you mind doing so, so you can get credit?

**Comment by tlinkowski** [comment]
Sure!

**Comment by tlinkowski** [comment]
Hm, I just realized that I implemented it inside [#305] (in the squashed commit 61e75370). Unfortunately, I forgot to add "[#308] more efficient Seq.splitAt implementation" to the commit message.

**Comment by lukaseder** [comment]
Oh yes indeed, thanks very much. The test also runs. I've activated it again.
